### ISSUE TITLE
[a11y]: Screen reader announces list item name three times in Customize Column tearsheet

### ISSUE THREAD

**Issue by tm-chi** [carbon-design-system/ibm-products#7343]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

v2.65.0-rc.0

### Automated testing tool and ruleset

manual

### Assistive technology

MacOS Voice Over

### Description

When opening Customize Column tearsheet from datagrid component and navigating to a list item by using the keyboard, the screen reader announces the item's name three times. The behavior might be distracting and reduce the clarity of the interaction.

<img width="1158" alt="Image" src="https://github.com/user-attachments/assets/8d6cddd8-cb45-454c-b00e-4f7e3d4b74f9" />

### Reproduction/example

https://ibm-products.carbondesignsystem.com/iframe.html?args=&globals=viewport%3Abasic&id=deprecated-datagrid-datagrid-columncustomization--column-customization-usage-story&viewMode=story

### Steps to reproduce

1. Select the `Customize columns` button from Datagrid's toolbar to open Customize columns tearsheet
2. Navigate a list item using keyboard

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by elycheea** [comment]
Thanks for reporting this! The Datagrid is currently deprecated in favor of [`TanStack/Table`](https://github.com/carbon-design-system/tanstack-carbon), which unfortunately means support for bugs and enhancements is limited. If you’d like to [contribute a fix](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CONTRIBUTING.md), I’m happy to reopen this issue for you.

If your team is unable to migrate to TanStack Table at this time, I highly recommend contributing a fix. If you’re interesting in learning more about the move to TanStack Table and migration experience, you’re welcome to join our [office hours](https://ec.yourlearning.ibm.com/w3/event/10482832). (You can find us in the IBM Products breakout. The Carbon React team can direct you from the main room!)
